base/libgit2/callbacks.jl

```diff
@@ -59,6 +59,8 @@ function credentials_callback(cred::Ptr{Ptr{Void}}, url_ptr::Cstring,
     # parse url for schema and host
     urlparts = match(urlmatcher, url)
     schema = urlparts.captures[1]
+    urlusername = urlparts.captures[4]
```

Comment:
Sorry, not sure what you're suggesting. The `urlusername` only ever gets used by https. Are you saying that libgit2 is already extracting this, so we should just load it from `username_ptr`?